File: src/tests/bench/bench_fib_ai.js
function fibRec(n){ if(n<=1) return n; return fibRec(n-1)+fibRec(n-2); }
function fibIter(n){ let a=0,b=1; for(let i=0;i<n;i++){ let t=a; a=b; b=t+b; } return a; }
function dot(v1,v2){ return v1.reduce((s,x,i)=>s+x*v2[i],0); }
function norm(v){ return Math.sqrt(v.reduce((s,x)=>s+x*x,0)); }
function softmax(v){ let e=v.map(Math.exp); let s=e.reduce((a,b)=>a+b,0); return e.map(x=>x/s); }

let v1=[1,2,3,4,5], v2=[5,4,3,2,1];

let start=Date.now(); let r1=fibRec(25); let t1=Date.now()-start;
start=Date.now(); let r2=fibIter(35); let t2=Date.now()-start;
start=Date.now(); let dot_val=dot(v1,v2); let norm_val=norm(v1); let sm=softmax(v1); let t3=Date.now()-start;

console.log(`fibRec(25) = ${r1} | tempo = ${t1} ms`);
console.log(`fibIter(35) = ${r2} | tempo = ${t2} ms`);
console.log(`dot(v1,v2) = ${dot_val}`);
console.log(`norm(v1) = ${norm_val}`);
console.log(`softmax(v1) = ${sm} | tempo = ${t3} ms`);

File: src/tests/bench/bench_fib_ai.py
import time
import math

# Fibonacci recursivo
def fib_rec(n):
    if n <= 1: return n
    return fib_rec(n-1) + fib_rec(n-2)

# Fibonacci iterativo
def fib_iter(n):
    a,b = 0,1
    for _ in range(n):
        a,b = b,a+b
    return a

# Dot, norm, softmax
v1 = [1.0,2.0,3.0,4.0,5.0]
v2 = [5.0,4.0,3.0,2.0,1.0]

start = time.perf_counter()
r1 = fib_rec(25)
t1 = (time.perf_counter() - start) * 1000

start = time.perf_counter()
r2 = fib_iter(35)
t2 = (time.perf_counter() - start) * 1000

start = time.perf_counter()
dot_val = sum(a*b for a,b in zip(v1,v2))
norm_val = math.sqrt(sum(a**2 for a in v1))
exp_v = [math.exp(x) for x in v1]
s = sum(exp_v)
softmax = [x/s for x in exp_v]
t3 = (time.perf_counter() - start) * 1000

print("fib_rec(25) =", r1, "| tempo =", round(t1,2),"ms")
print("fib_iter(35) =", r2, "| tempo =", round(t2,2),"ms")
print("dot(v1,v2) =", dot_val)
print("norm(v1) =", norm_val)
print("softmax(v1) =", softmax, "| tempo =", round(t3,2),"ms")